bodo/tests/test_df_lib/test_end_to_end.py

```diff
@@ -425,7 +429,34 @@ def test_head_pushdown(datapath):
     )
 
 
-@pytest.mark.skip(reason="Non-pushdown physical limit node needs work.")
+@pytest.mark.skip(reason="Not working.")
+def test_projection_head_pushdown(datapath):
+    """Test for head pushed down to read parquet."""
+    bodo_df1 = bd.read_parquet(datapath("dataframe_library/df1.parquet"))
+    bodo_df2 = bodo_df1["D"]
+    bodo_df3 = bodo_df2.head(3)
+
+    # Make sure bodo_df2 is unevaluated at this point.
+    assert bodo_df3.is_lazy_plan()
+    assert bodo_df3.plan is not None
+
+    # pre, post = bd.utils.getPlanStatistics(bodo_df3.plan)
+    # _test_equal(pre, 3)
+    # _test_equal(post, 2)
+
+    py_df1 = pd.read_parquet(datapath("dataframe_library/df1.parquet"))
+    py_df2 = py_df1["D"]
+    py_df3 = py_df2.head(3)
+
+    _test_equal(
+        bodo_df3,
+        py_df3,
+        check_pandas_types=False,
+        sort_output=True,
+        reset_index=True,
+    )
```

Comment:
The Pandas `head` result won't be equal.
```suggestion
```

Reply:
done